apisix/init.lua

```diff
@@ -823,6 +823,11 @@ function _M.stream_init_worker()
         core.config.init_worker()
     end
 
+    local discovery = require("apisix.discovery.init").discovery
+    if discovery and discovery.init_worker then
+        discovery.init_worker()
+    end
```

Comment:
You can add a test with the DNS discovery: https://github.com/apache/apisix/blob/07bfe5b9c03ad361cb71541eb02fd661f1323213/t/discovery/dns/sanity.t#L91

You can setup it with https://github.com/apache/apisix/blob/07bfe5b9c03ad361cb71541eb02fd661f1323213/utils/set-dns.sh#L33